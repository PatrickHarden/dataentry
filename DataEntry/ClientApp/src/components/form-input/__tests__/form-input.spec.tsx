import React from 'react';
import { mount, shallow } from 'enzyme';

import FormInput from '../form-input';

describe('components', () => {

  describe('form-input', () => {

    let props:any;
    let wrapper:any;

    beforeEach(() => {
        props = {
            name: "testme",
            defaultValue: "Some Value",
            label: "Test Me!",
            error: false,
            blurHandler: jest.fn()
        }
    });

    // it renders it's normal state properly (border should be #dedede)
    it("normal state shows normal border (#cccccc)", () => {
        wrapper = mount(<FormInput {...props}/>);
        expect(wrapper.find('input')).toHaveStyleRule("border","1px solid #cccccc");
    })

    // it renders it's error state properly (border should be red)
    it("error state shows red border", () => {
        props = {
            name: "testme",
            defaultValue: "Some Value",
            label: "Test Me!",
            error: true,
            blurHandler: jest.fn()
        }
        wrapper = mount(<FormInput {...props}/>);
        expect(wrapper.find('input')).toHaveStyleRule("border","1px solid red");
    })

    // it shows its normal border
    it("when not in an error state, it shows a normal border", () => {
        wrapper = mount(<FormInput {...props}/>);
        expect(wrapper.find('input')).toHaveStyleRule("border","1px solid #cccccc");
    })

    // it has the correct default value set
    it("has the correct default value", () => {
        wrapper = mount(<FormInput {...props}/>);
        expect(wrapper.find('input').instance().value).toEqual("Some Value");
    })
    
    // it has a label of "Test Me!"
    it("renders the expected label", () => {
        wrapper = mount(<FormInput {...props}/>);
        expect(wrapper.find('h5').text()).toEqual("Test Me!");
    });

    // snapshot render
    it('renders according to snapshot', () => {
        expect(shallow(<FormInput {...props} />)).toMatchSnapshot();
    });
  });
});