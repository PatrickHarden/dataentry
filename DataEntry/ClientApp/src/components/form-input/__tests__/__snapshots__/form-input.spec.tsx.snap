// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components form-input renders according to snapshot 1`] = `
<styled.div>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <FormFieldLabel
      error={false}
      label="Test Me!"
      styles={Object {}}
    />
  </div>
  <styled.input
    defaultValue="Some Value"
    error={false}
    id="testme"
    name="testme"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    onPaste={[Function]}
    tabIndex={0}
  />
</styled.div>
`;
